src/orchestrator/core/mongo.py

```diff
@@ -58,6 +59,20 @@ def createIndexes(self, SERVICE_NAME):
         except Exception, e:
             logger.warn("createIndex database %s exception: %s" % (databaseName,e))
 
+
+    def createSTHIndexes(self, SERVICE_NAME, SUBSERVICE_NAME):
+        try:
+            # For STH
+            databaseName = 'sth_' + SERVICE_NAME
+            db = self.client[databaseName]
+            collectionName = 'sth_/' + SUBSERVICE_NAME
```

Comment:
Maybe it should be  `collectionName = 'sth_/' + SUBSERVICE_NAME + '.aggr'` ?

In addition, it seems the code is creating four one-key indexes. However, what should be done is create one four-keys index.

Reply:
Both fixed in https://github.com/telefonicaid/orchestrator/pull/204/commits/1507dbcd7bc44d714e34a5bb6d3741d9c32880c7

Reply:
One-key indexes is good for search by all fields, like `index _id.entityId, _id.attrName, _id.resolution, _id.origin` but I'm not sure if it's optimized when you search just by entityId of attrName. Are we sure that the most common query search included all that fields?

Reply:
As far as I checked using MongoDB profiler, the query (applied in a $match statement in an aggregation pipeline is this):

```
                                "$match" : {
                                        "_id.entityId" : "NativeModel.50043034",
                                        "_id.entityType" : "device",
                                        "_id.attrName" : "batteryCharge",
                                        "_id.resolution" : "hour",
                                        "_id.origin" : {
                                                "$lte" : ISODate("2019-10-26T00:00:00Z"),
                                                "$gte" : ISODate("2018-10-16T00:00:00Z")
                                        }
                                }
```

so I'm pretty sure.

Reply:
Ok. NTC